railties/lib/rails/test_help.rb

```diff
@@ -44,3 +44,7 @@ def before_setup # :nodoc:
     super
   end
 end
+
+if defined?(Rails.application) && Rails.application && !Rails.application.task_loaded?
```

Comment:
There is no problem with `before_setup` as it will not be called until those classes are actually used.

This file is required not only by Rails application and Rails engine but also by Rails plugins. 
In Rails plugins, `Rails.application may not exist.  
(If using the dummy application, `Rails.application` is there, so in many cases I think there is no problem, but I am not sure ...) 

Also, I noticed that checking with `Minitest.run_via` was good, so I responded with that.